### PR TITLE
Feature etb#228 afec hwas seds driver

### PR DESCRIPTION
**Pull request by jrachucki-n7s**

### CODE REVIEW

File: Drivers/seds/AfecHwas/example/work/hwas/implem/default/C/src/hwas.c [hunk 1]
```diff
@@ -0,0 +1,334 @@
+/* Body file for function HWAS
+ * Generated by TASTE on 2021-11-29 11:46:27
```
Comment (Lurkerpas) [inline Drivers/seds/AfecHwas/example/work/hwas/implem/default/C/src/hwas.c]: Let's remove this.

File: Drivers/seds/AfecHwas/example/work/manager/implem/default/C/src/manager.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * Generated by TASTE on 2022-04-06 11:12:12
+ * You can edit this file, it will not be overwritten
+ * Provided interfaces : GetTemperature
+ * Required interfaces : AfecHwasInterface_GetValue_Pi, AfecHwasInterface_InitInstance_Pi
+ * User-defined properties for this function:
+ * Timers              : 
+ */
+
+#include "manager.h"
+//#include <stdio.h>
```
Comment (Lurkerpas) [inline Drivers/seds/AfecHwas/example/work/manager/implem/default/C/src/manager.c]: Remove

File: Drivers/seds/AfecHwas/example/work/manager/implem/default/C/src/manager.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+void manager_PI_GetTemperature(void)
+{
+   // Write your code here
+    if(!initialized)
+    {
+        manager_RI_AfecHwasInterface_InitInstance_Pi(&afec, &config);
+        initialized = true;
+    }else {
+        manager_RI_AfecHwasInterface_GetValue_Pi(&afec, &channelNumber, &tempRawValue);
```
Comment (Lurkerpas) [inline Drivers/seds/AfecHwas/example/work/manager/implem/default/C/src/manager.c:32]: How is the reading checked?

File: Drivers/seds/AfecHwas/example/work/manager/implem/default/C/src/manager.c (hunk 2 above)
Comment (jrachucki-n7s) [reply Drivers/seds/AfecHwas/example/work/manager/implem/default/C/src/manager.c]: Ok, I think this is automatically added by space-creator. (this is a part of the template)

File: Drivers/seds/AfecHwas/example/work/manager/implem/default/C/src/manager.c (hunk 3 above)
Comment (jrachucki-n7s) [reply Drivers/seds/AfecHwas/example/work/manager/implem/default/C/src/manager.c:32]: I am checking it while debugging. I want to sent this value over UART in the UART demo

File: Drivers/seds/AfecHwas/example/work/hwas/implem/default/C/src/hwas.c (hunk 1 above)
Comment (jrachucki-n7s) [reply Drivers/seds/AfecHwas/example/work/hwas/implem/default/C/src/hwas.c]: Done

File: Drivers/seds/AfecHwas/example/work/manager/implem/default/C/src/manager.c (hunk 2 above)
Comment (jrachucki-n7s) [reply Drivers/seds/AfecHwas/example/work/manager/implem/default/C/src/manager.c]: done

File: Drivers/seds/AfecHwas/AfecHwas.xml [hunk 4]
```diff
@@ -0,0 +1,562 @@
...
+                  </EndAt>
+                  <Do>
+                    <SendCommandPrimitive shortDescription="Get analogChannel status register value" interface="RawMemoryAccess" command="ReadWord">
+                      <ArgumentValue name="address">
+                        <VariableRef variableRef="sourceAddressVar" />
+                      </ArgumentValue>
+                      <ArgumentValue name="mask">
+                        <VariableRef variableRef="channelMaskVar" />
+                      </ArgumentValue>
+                      <ArgumentValue name="inputValue">
```
Comment (Lurkerpas) [inline Drivers/seds/AfecHwas/AfecHwas.xml]: outputValue?

File: Drivers/seds/AfecHwas/AfecHwas.xml [hunk 5]
```diff
@@ -0,0 +1,562 @@
...
+                  </ArgumentValue>
+                  <ArgumentValue name="inputValue">
+                    <VariableRef variableRef="bitsToSetVar" />
+                  </ArgumentValue>
+                </SendCommandPrimitive>
+                <Assignment outputVariableRef="sourceAddressVar">
+                  <LongDescription>Assign destination register address</LongDescription>
+                  <VariableRef variableRef="afecVar.mInstanceChsrAddress"></VariableRef>
+                </Assignment>
+                <Iteration iteratorVariableRef="readBitsVar">
```
Comment (Lurkerpas) [inline Drivers/seds/AfecHwas/AfecHwas.xml:427]: I think this is an overinterpretation of SEDS spec. The proper way to do such wait should be via onTimer transition between discrete states (like waiting). I will add task to Jira to change this later in order not to stall. Please add a comment indicating a TODO.

File: Drivers/seds/AfecHwas/AfecHwas.xml (hunk 4 above)
Comment (jrachucki-n7s) [reply Drivers/seds/AfecHwas/AfecHwas.xml]: fixed

File: Drivers/seds/AfecHwas/AfecHwas.xml (hunk 5 above)
Comment (jrachucki-n7s) [reply Drivers/seds/AfecHwas/AfecHwas.xml:427]: I tried to make it look similar to waiting in a while loop until proper bits in a given register will be set.

Comment (Lurkerpas) [reply Drivers/seds/AfecHwas/AfecHwas.xml:427]: OK... I still don't see the todo, but maybe task on Jira will suffice... in fact, it should be better. Discussion was held, problem understood.

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**